src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs

```diff
@@ -55,6 +57,11 @@ internal SourceEventSymbol(
             bool modifierErrors;
             _modifiers = MakeModifiers(modifiers, isExplicitInterfaceImplementation, isFieldLike, _location, diagnostics, out modifierErrors);
             this.CheckAccessibility(_location, diagnostics, isExplicitInterfaceImplementation);
+
+            if (IsPartial)
+            {
+                ModifierUtils.CheckFeatureAvailabilityForPartialEventsAndConstructors(_location, diagnostics);
```

Comment:
nit: it seems a little inconsistent that feature availability is checked in constructor for event symbols but during modifier checks for constructor symbols.